from database import Database
class UserModel:
    def __init__(self, db: Database):
        self.db = db
        self.conn = self.db.get_connection()

    def authenticate(self, username: str, password: str) -> bool:

        if not self.conn:
            raise Exception("Lỗi kết nối cơ sở dữ liệu")
        try:
            cursor = self.conn.cursor()
            query = "SELECT * FROM taikhoan WHERE tai_khoan = %s AND mat_khau = %s"
            cursor.execute(query, (username, password))
            user = cursor.fetchone()
            cursor.close()
            return user is not None
        except Exception as e:
            raise Exception(f"Lỗi trong quá trình xác thực: {e}")

    def getAllAccounts(self):

        try:
            cursor = self.conn.cursor()
            query = "SELECT nv.ma_nhan_vien, nv.ho_va_ten, tk.tai_khoan, tk.mat_khau, tk.vai_tro, tk.trang_thai_tai_khoan from taikhoan tk INNER JOIN nhanvien nv on nv.id = tk.id_nhan_vien"
            cursor.execute(query)
            result = cursor.fetchall()
            self.conn.close()
            return result
        except:
            raise Exception